[4657] Reactify Global App (Groups)

aachavarria opened the pull request:
craftercms/craftercms#4657

aachavarria committed:
Creating GroupEditDialog.tsx
Creating GroupsGridUI.tsx;
Updating GroupsManagement.tsx;

aachavarria committed:
Creating TransferList.tsx;
Adding TransferList.tsx to GroupEditDialogUI.tsx

aachavarria committed:
Merge branch '4.0.0/develop' of https://github.com/rart/studio-ui into feature/global-menu/groups-reactify

 Conflicts:
	ui/app/src/components/GroupsManagement/GroupsManagement.tsx

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/global-menu/groups-reactify

rart commented on ui/app/src/components/GroupEditDialog/styles.ts:
```diff
@@ -0,0 +1,58 @@
+
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+
+export const useStyles = makeStyles((theme) =>
```
export default below

rart commented on ui/app/src/components/GroupsGrid/styles.ts:
```diff
@@ -0,0 +1,33 @@
+
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+
+export const styles = makeStyles((theme) =>
```
export default below

rart commented on ui/app/src/components/GroupsManagement/GroupsManagement.tsx:
```diff
@@ -31,6 +97,32 @@ export default function GroupsManagement() {
+          }}
+          withEmptyStateProps={{
+            emptyStateProps: {
+              title: <FormattedMessage id="usersGrid.emptyStateMessage" defaultMessage="No Users Found" />
```
Did you mean groups?

aachavarria committed:
TransferList.tsx refactors;

aachavarria committed:
UserInfoDialogUI.tsx: updating labels;
Creating deleteUsersFromGroup;
Updating onTargetListItemsAdded, onTargetListItemsRemoved on TransferList.tsx;
Updating GroupEditDialogUI.tsx to use Labels;
GroupEditDialogContainer.tsx: integrating onRemoveMembers, onAddMembers;

aachavarria committed:
adding setDirty, and updating group;

aachavarria committed:
Adding GroupsManagement.tsx to legacy;
Creating CreateGroupDialog.tsx;

aachavarria committed:
Deleting CreateGroupDialog.tsx;
Refactors on GroupEditDialog.tsx to support Create;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/global-menu/groups-reactify

 Conflicts:
	ui/app/src/styles/theme.tsx

aachavarria committed:
Renaming GroupEditDialog.tsx to EditGroupDialog.tsx;

rart commented on ui/app/src/components/EditGroupDialog/EditGroupDialog.tsx:
```diff
@@ -0,0 +1,39 @@
+
+export interface EditGroupDialogProps {
+  open: boolean;
+  group: Group;
```
Isn't this supposed to be optional?

rart commented on ui/app/src/components/TransferList/TransferList.tsx:
```diff
@@ -0,0 +1,208 @@
+  return (
+    <Paper className={classes.listPaper}>
+      <header className={classes.listHeader}>
+        <Checkbox checked={isAllChecked(items)} onChange={(event) => onCheckAllClicked(items, event.target.checked)} />
```
Move to own file **and directory**

rart commented on ui/app/src/components/TransferList/styles.ts:
```diff
@@ -0,0 +1,49 @@
+      overflow: 'auto'
+    }
+  })
+);
```
Split TransferList vs TransferListColumn styles

rart commented on ui/app/src/components/TransferList/TransferList.tsx:
```diff
@@ -0,0 +1,208 @@
+export interface TransferListItem {
+  id: string;
+  title: string;
+  subTitle?: string;
```
```suggestion
  subtitle?: string;
```

rart commented on ui/app/src/components/TransferList/TransferList.tsx:
```diff
@@ -0,0 +1,208 @@
+      <header className={classes.listHeader}>
+        <Checkbox checked={isAllChecked(items)} onChange={(event) => onCheckAllClicked(items, event.target.checked)} />
+        {title && <Typography color="textSecondary">{title}</Typography>}
+        <SearchBar keyword="" onChange={() => {}} classes={{ root: classes.searchBar }} />
```
onChange doesn't do anything. the SearchBar doesn't do anything.

aachavarria replied:
that is true the search is not implemented yet

rart commented on ui/app/src/services/groups.ts:
```diff
@@ -29,14 +29,14 @@ const paginationDefault = {
 };
 
 export function fetchAll(options?: PaginationOptions): Observable<PagedArray<Group>> {
```
```suggestion
export function fetchAll(options?: Partial<PaginationOptions>): Observable<PagedArray<Group>> {
```

rart commented on ui/app/src/components/TransferList/TransferList.tsx:
```diff
@@ -0,0 +1,208 @@
+}
+
+export interface TransferListItem {
+  id: string;
```
```suggestion
  id: string | number;
```

rart commented on ui/app/src/components/TransferList/TransferList.tsx:
```diff
@@ -0,0 +1,208 @@
+interface TransferListProps {
+  source: TransferListObject;
+  target: TransferListObject;
+  inProgressIds: string[];
```
```suggestion
  inProgressIds: Array<string | number>;
```

aachavarria committed:
PR Feedback;
Refactor on TransferList.tsx;
Creating TransferListColumn.tsx;

rart commented on ui/app/src/components/TransferList/TransferList.tsx:
```diff
@@ -0,0 +1,208 @@
+        {items.map((item, i) => (
+          <ListItem
+            disabled={inProgressIds.includes(item.id)}
+            key={i}
```
**Make sure to carry this change when you move component to it's file**
```suggestion
            key={item.id}
```

rart commented on ui/app/src/components/TransferList/TransferList.tsx:
```diff
@@ -0,0 +1,208 @@
+      <header className={classes.listHeader}>
+        <Checkbox checked={isAllChecked(items)} onChange={(event) => onCheckAllClicked(items, event.target.checked)} />
+        {title && <Typography color="textSecondary">{title}</Typography>}
+        <SearchBar keyword="" onChange={() => {}} classes={{ root: classes.searchBar }} />
```
Align SearchBar Right.

rart commented on ui/app/src/components/TransferList/TransferList.tsx:
```diff
@@ -0,0 +1,208 @@
+    <Paper className={classes.listPaper}>
+      <header className={classes.listHeader}>
+        <Checkbox checked={isAllChecked(items)} onChange={(event) => onCheckAllClicked(items, event.target.checked)} />
+        {title && <Typography color="textSecondary">{title}</Typography>}
```
Make sure to test with a long title. How does title+searcbar behave in that case?

aachavarria committed:
PR Feedback;
Refactor on TransferList.tsx;
Creating TransferListColumn.tsx;

aachavarria committed:
PR Feedback;

aachavarria committed:
PR Feedback;

aachavarria committed:
Adding emptyStateMessage prop to TransferListColumn.tsx;
Updates on TransferList.tsx;
PR Feedback;

rart commented on ui/app/src/components/EditGroupDialog/EditGroupDialog.tsx:
```diff
@@ -0,0 +1,39 @@
+
+import React from 'react';
+import Group from '../../models/Group';
+import { Dialog } from '@material-ui/core';
```
```suggestion
import Dialog from '@material-ui/core/Dialog';
```

rart commented on ui/app/src/components/TransferListColumn/TransferListColumn.tsx:
```diff
@@ -0,0 +1,120 @@
+import EmptyState from '../SystemStatus/EmptyState';
+import { FormattedMessage } from 'react-intl';
+
+interface TransferListColumnProps {
```
export

rart commented on ui/app/src/services/groups.ts:
```diff
@@ -68,18 +68,37 @@ export function update(group: Partial<Group>): Observable<Group> {
-export function addUsersToGroup(id: number, users: { ids: string[]; usernames: string[] }): Observable<User[]> {
-  return postJSON(`/studio/api/2/groups/${id}/members`, users).pipe(pluck('response', 'users'));
+export function addUserToGroup(groupId: number, username: string): Observable<User[]> {
+  return postJSON(`/studio/api/2/groups/${groupId}/members`, [username]).pipe(pluck('response', 'users'));
```
```suggestion
export function addUserToGroup(groupId: number, username: string): Observable<User> {
  return addUsersToGroup(groupId, [username]).pipe(pluck(0));
```

rart commented on ui/app/src/services/groups.ts:
```diff
@@ -68,18 +68,37 @@ export function update(group: Partial<Group>): Observable<Group> {
+export function addUsersToGroup(groupId: number, users: { usernames: string[] }): Observable<User[]>;
+export function addUsersToGroup(groupId: number, users: { ids?: number[]; usernames?: string[] }): Observable<User[]> {
+  return postJSON(`/studio/api/2/groups/${groupId}/members`, users).pipe(pluck('response', 'users'));
 }
```
```suggestion
export function addUsersToGroup(groupId: number, ids: number[]): Observable<User[]>;
export function addUsersToGroup(groupId: number, usernames: string[]): Observable<User[]>;
export function addUsersToGroup(groupId: number, idsOrUsernames: Array<number> | Array<string>): Observable<User[]> {
  return postJSON(`/studio/api/2/groups/${groupId}/members`, {
    [typeof idsOrUsernames[0] === 'string' ? 'usernames' : 'ids']: idsOrUsernames
  }).pipe(pluck('response', 'users'));
}
```

rart commented on ui/app/src/services/groups.ts:
```diff
@@ -68,18 +68,37 @@ export function update(group: Partial<Group>): Observable<Group> {
   const qs = toQueryString({
     userId,
     username
   });
```
```suggestion
export function deleteUserFromGroup(groupId: number, username: string): Observable<true>;
export function deleteUserFromGroup(groupId: number, userId: number): Observable<true>;
export function deleteUserFromGroup(groupId: number, usernameOrUserId: number | string): Observable<true> {
```

rart commented on ui/app/src/services/groups.ts:
```diff
@@ -68,18 +68,37 @@ export function update(group: Partial<Group>): Observable<Group> {
     username
   });
-  return del(`/studio/api/2/groups/${id}/members${qs}`).pipe(mapTo(true));
+  return del(`/studio/api/2/groups/${groupId}/members${qs}`).pipe(mapTo(true));
```
```suggestion
  // @ts-ignore - types are correct, signature to accept either is simply not published by deleteUsersFromGroup
  return deleteUsersFromGroup(groupId, [usernameOrUserId]);
```

rart commented on ui/app/src/services/groups.ts:
```diff
@@ -68,18 +68,37 @@ export function update(group: Partial<Group>): Observable<Group> {
+    username: user.usernames
+  });
+  return del(`/studio/api/2/groups/${groupId}/members${qs}`).pipe(mapTo(true));
 }
```
```suggestion
export function deleteUsersFromGroup(groupId: number, userIds: number[]): Observable<true>;
export function deleteUsersFromGroup(groupId: number, usernames: string[]): Observable<true>;
export function deleteUsersFromGroup(groupId: number, usernamesOrUserIds: number[] | string[]): Observable<true> {
  const qs = toQueryString({
    [typeof usernamesOrUserIds[0] === 'string' ? 'username' : 'userId']: usernamesOrUserIds
  });
  return del(`/studio/api/2/groups/${groupId}/members${qs}`).pipe(mapTo(true));
}
```

aachavarria committed:
Refactors on groups.ts service;

aachavarria committed:
GroupsGridUI.tsx removing classes.root to use Box;

rart commented on ui/app/src/components/TransferList/TransferList.tsx:
```diff
@@ -0,0 +1,181 @@
+import TransferListColumn from '../TransferListColumn';
+import { FormattedMessage } from 'react-intl/dist/react-intl';
+
+interface TransferListProps {
```
Export

rart commented on ui/app/src/components/GroupsGrid/GroupsGridUI.tsx:
```diff
@@ -0,0 +1,86 @@
+            </GlobalAppGridRow>
+          </TableHead>
+          <TableBody>
+            {groups?.map((group, i) => (
```
Why would this ever be null?

rart commented on ui/app/src/components/GroupsManagement/GroupsManagement.tsx:
```diff
@@ -14,23 +14,131 @@
           </Button>
         }
       />
+      <ErrorBoundary>
```
Remove

aachavarria committed:
PR Feedback;

aachavarria committed:
Fixing incorrect import for FormattedMessage
